thrust/detail/pointer.h

```diff
@@ -178,7 +179,13 @@ template<typename Element, typename Tag, typename Reference, typename Derived>
     Element *get() const;
 }; // end pointer
 
+// Output stream operator
+template<typename Element, typename Tag, typename Reference, typename Derived,
+         typename charT, typename traits>
+std::basic_ostream<charT, traits> &
+operator<<(std::basic_ostream<charT, traits> &os,
```

Comment:
It already exists for `thrust::device_ptr`, this simply promotes it to `thrust::pointer`.